apps/rush-lib/src/cli/actions/InstallAction.ts

```
@@ -6,128 +6,87 @@ import * as os from 'os';
-
-      if (!shrinkwrapFile) {
-        console.log('');
-        console.log(colors.red('Unable to proceed: The shrinkwrap file is missing.'));
-        console.log('');
-        console.log('You need to run "rush generate" first.');
-        return process.exit(1);
-      }
-
-      let installType: InstallType = InstallType.Normal;
-      if (this._cleanInstallFull.value) {
-        installType = InstallType.UnsafePurge;
-      } else if (this._cleanInstall.value) {
-        installType = InstallType.ForceClean;
-      }
-
-      if (!installManager.createTempModulesAndCheckShrinkwrap(
-        shrinkwrapFile,
-        installType !== InstallType.Normal
-      )) {
-        console.log('');
-        console.log(colors.red('You need to run "rush generate" to update your shrinkwrap file.'));
-        return process.exit(1);
-      }
-
-      installManager.installCommonModules(installType);
-    }).catch((error) => {
-      stopwatch.stop();
-      this._collectTelemetry(stopwatch, false);
-      throw error;
-    }).then(() => {
-      stopwatch.stop();
-      console.log(colors.green(`Done. (${stopwatch.toString()})`));
-
-      this._collectTelemetry(stopwatch, true);
-
-      this.eventHooksManager.handle(Event.postRushInstall);
-
-      if (!this._noLinkParameter.value) {
-        const linkManager: BaseLinkManager = LinkManagerFactory.getLinkManager(this.rushConfiguration);
-        return linkManager.createSymlinksForProjects(false);
-      } else {
-        console.log(os.EOL + 'Next you should probably run: "rush link"');
-      }
-    });
+        this._collectTelemetry(stopwatch, true);
+        this.eventHooksManager.handle(Event.postRushInstall);
+
+        if (success) {
+          console.log(os.EOL + colors.green(`Rush install finished successfully. (${stopwatch.toString()})`));
+        } else {
+          console.log(os.EOL + `Rush install completed. (${stopwatch.toString()})`);
+        }
+
+      });
```

>}); [](start = 6, length = 3)

You'd want to throw an exception to get the exitCode.

---

I think you misunderstood the model.

---
In reply to: [187452722](https://github.com/Microsoft/web-build-tools/pull/646#discussion_r187452722) [](ancestors = 187452722)